### ISSUE TITLE
weird mouse event behaviour

### ISSUE THREAD

**Issue by blancoberg** [surge-synthesizer/surge#7942]
**Bug Description:**
When a context menu is open ( right click ) and you switch to another window, a mouse down event seem to be randomly called when you hover the mouse on top of surge causing all sorts of behaviours from moving/resizing the window, dragging sliders etc.. 

**Surge XT Version**
Nightly 1.4

* Version:
Standalone

**Reproduction Steps:**
1. Open the context menu for one of the sliders.
2. switch to another window/app
3.  hover the mouse over surge xt.
4.  a mouse down event is called ( sometimes )

**Expected Behavior:**
No mouse event is called :)

**Computer Information (please complete the following!):**
Windows 10

https://github.com/user-attachments/assets/cf34ef69-bc40-45b1-bab8-a15988384134

**Comment by blancoberg** [comment]
likely a bug in juce::PopupMenu(). its like it doesnt get destroyed properly when window is set out of focus

**Comment by baconpaul** [comment]
there's a dismiss all popups call. we call it in the destructor of SurgeGUIEditor
We probably also need to call it when focus is lost on SGE
but that conditions a bit tricky to find

**Comment by landinig** [comment]
I can confirm some odd behaviour when resizing the main window (under linux). Intermittently, the resizing seems to remain "on" and attempting to place the cursor over the synth window remains resizing again making it smaller every time.

**Comment by landinig** [comment]
Hi @blancoberg, sorry I am not sure, I have just been using the program for a couple of days. It seems that when I drag, let's say the 'top border' or 'top right corner' (which should put it in resizing mode) sometimes one cannot exit the resizing state and attempt to go into the main panel keeps resizing it (smaller each time!) until it completely minimises. 
In case it is useful:
Version: Surge XT 1.3.4.f7b97c6
Build Info: 2024-08-11 @ 15:57:53 on '24d2f7d926f1/local' with 'GNU-9.4.0' using JUCE 7.0.12
System Info: Linux 64-bit Standalone on  Intel(R) Core(TM) i7-4790 CPU @ 3.60GHz, 31 GB RAM
Sample Rate: 44100 Hz
Processing Block: 32 samples

**Comment by landinig** [comment]
@blancoberg Another comment: comparing to how KDE works,  the resizing state is held while the left mouse button is pressed. It seems that in Surge-XT resizing works a toggle state ? (i.e. releasing the button does not stop the resizing).

**Comment by mkruselj** [comment]
I cannot seem to reproduce this one on my W10 install with the reproduction steps provided...

**Comment by blancoberg** [comment]
I just replicated it in win11. Its still there. 
right click on of the oscillator sliders, click on another open window to give it focus and quickly hover around over surge xt. it should produce a mousedown event. 

You have to be somewhat fast as it appears to be triggered after a set time.  100-300 milliseconds or so

**Comment by mkruselj** [comment]
Yeah I'm doing it with Alt+Tab to focus on another window and the menu just disappears and when hovering over Surge I see no mouse down event happening...

**Comment by blancoberg** [comment]
you have to click on another window to give it focus, It doesnt seem to happen when you alt+tab

**Comment by mkruselj** [comment]
Tried that too, still nothing...

**Comment by mkruselj** [comment]
@blancoberg I wonder if this is down to which mouse we might be using... I still cannot reproduce this. Logitech MX Master 2S over here.

**Comment by blancoberg** [comment]
maybe,. I'll see if I can try a different mouse

**Comment by mkruselj** [comment]
@blancoberg Did you try with another mouse? :)

**Comment by nuoun** [comment]
So I can't reproduce this either but I have had the same class of glitch happen a few times on KDE and XFCE where the resize window state just got stuck somehow. Wonder if crossing screen boundaries and/or a high DPI mouse might have anything to do with it.

**Comment by mkruselj** [comment]
Pushing out of 1.4 milestone, not a dealbreaker for its release. Plus, I have not been able to reproduce it at all...

**Comment by mkruselj** [comment]
Joel reports that he cannot reproduce this issue anymore on his W11 install, with nightly and 1.3.4 since he got a new mouse. That seems like the smoking gun here. So, closing the issue!
